crates/ruma-common/src/events.rs

```diff
@@ -197,12 +197,16 @@ pub use self::{
 
 /// The base trait that all event content types implement.
 ///
-/// Implementing this trait allows content types to be serialized as well as deserialized.
+/// Use [`macros::EventContent`] to derive this traits. It is not meant to be implemented manually.
 pub trait EventContent: Sized + Serialize {
-    /// A matrix event identifier, like `m.room.message`.
+    /// The Rust enum for the event kind's known types.
+    type EventType;
```

Comment:
I like these as associated types, is this to help with whatever the extensible events stuff is? Like now it's easier to have `m.whatever.you.want`?

Reply:
This is a refactoring I've wanted to do to enable the change you see in client-api in a straight-forward way. This whole PR is also preparation for variable event types (#686) which will touch lots of the same code and also breaks our previous assumption that we can borrow the event type from the event (unless we were to store the full thing as a field for variable event type content structs, but I don't really like that); using an owned `EventType` instead gets rid of that.